eng/common/scripts/git-branch-push.ps1

```diff
@@ -51,8 +51,19 @@ if ($LASTEXITCODE -ne 0)
     exit $LASTEXITCODE
 }
 
-Write-Host "git checkout -b $PRBranchName"
-git checkout -b $PRBranchName
+# 0 if found, false if not
+git show-ref --verify --quiet refs/heads/$PRBranchName
```

Comment:
/CC @weshaggard 
Two things
1. Why is this change necessary?
2. I'm not sure you need the refs/heads/$PRBranchName or if the $PRBranchName is sufficient.

Reply:
I'm accumulating changes in the `PR` branch. My understanding of this script is that it'll always attempt to check the branch as new in the current iteration. Unfortunately, when you fetch upstream, it also picks up that PR branch. After that, the git checkout -b will fatal with `branch already exists`. 

I added the change so that when it DOES find the branch I'm attempting to push to, it won't just crash out.

Hmm. I may be able to accumulate in smoke-test only and git-branch-push to the other branch. I'm not sure. WIll try it when I'm back at my machine.

Reply:
`git checkout -b $PRBranchName` here only create the branch in the local clone and should never fail on a fresh clone that hasn't already created that branch. If the branch already exists from upstream this command will track that upstream branch and make the branch pointers the same. So I don't think there is any need to check for this branch.

Reply:
@weshaggard That's what I thought and why I'd asked the question. If that had been a problem we'd have seen it during the last few releases as there are a number of pipelines that create add commits to the same branch. I was more curious as to whether or not @scbedd was seeing something going awry that was unique to his scenario.

Reply:
Ok. I'll try removing this piece and see what happens. I must be missing something because this absolutely failed on my box without the code change.

Reply:
feel free to ping me and I can explain why I think it failed locally but won't fail on ci machines. The biggest reason is locally I suspect you are re-using the clone which already has the branch created, but in ci it is always a fresh clone.

Reply:
Based on your other comments if you are trying to make multiple commits the the branch then I'd suggest you do all those commits in another temp branch before calling this script and then use this script to handle the pushing of the real branch for doing the pull request.

Reply:
Hmm. My goal here is to get zero-touch to the doc repo from each release. I just pushed a refactor that moves to a yml abstraction layer + does a readme per.

Reply:
Alrighty. This will be configured in each release.